godot-core/src/obj/raw_gd.rs

```diff
@@ -437,14 +448,18 @@ where
         }
 
         let callbacks = crate::storage::nop_instance_callbacks();
-        let token = sys::get_library() as *mut std::ffi::c_void;
-        let binding = interface_fn!(object_get_instance_binding)(self.obj_sys(), token, &callbacks)
-            as sys::GDExtensionClassInstancePtr;
+        // SAFETY: library is already initialized.
+        let token = unsafe { sys::get_library() };
+        // SAFETY: ensured that `self.obj` is non-null and valid.
+        let binding = unsafe {
+            interface_fn!(object_get_instance_binding)(self.obj_sys(), token.cast(), &callbacks)
+        };
```

Comment:
Please empty lines before "groups" of comment+code:
```suggestion

        // SAFETY: library is already initialized.
        let token = unsafe { sys::get_library() };

        // SAFETY: ensured that `self.obj` is non-null and valid.
        let binding = unsafe {
            interface_fn!(object_get_instance_binding)(self.obj_sys(), token.cast(), &callbacks)
        };
```

But more importantly, type inference for dangerous pointer conversions like `.cast()` is really an antipattern in my opinion. It also bears the risk that if the signature changes, then our conversion semantics are silently updated, which may be unintended. Not a big deal here because GDExtension APIs are stable, but still... Naming the type ensures that our mental model is correct.

Could you declare `token` with the explicit converted-to type, e.g.:
```rs
        // SAFETY: library is already initialized.
        let token = unsafe { sys::get_library() };
        let token = token.cast::<???>();
```

Reply:
If we are going to establish a convention, I would go with 

```rs
let variable: T = ptr.cast();
```

due to type aliases (as far as I'm aware `.cast::<*#$%>!@#(&!!()-| (choose the right one or none) TypeAlias>()` is not an option :thinking:? I've never seen it)

example:

```rs
(...)
pub type GDExtensionClassInstancePtr = *mut __GdextClassInstance;

// Silly – `sys::__GdextClassInstance` is not mentioned anywhere else, and we can't cast directly to `sys::GDExtensionClassInstancePtr`.
let ptr: sys::GDExtensionClassInstancePtr = binding.cast::<sys::__GdextClassInstance>();
```

Reply:
Makes sense, but there are also cases where you have the pointee type and would need to write an extra `*mut` or `*const` around it, if used on the left-hand-side.

We should probably just use whichever is easier, main thing is that a type is involved somewhere :slightly_smiling_face: